x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx

```diff
@@ -41,22 +98,301 @@ export function FieldItem({ field, indexPattern, highlight, exists }: FieldItemP
       </span>
     );
 
+  let formatter: { convert: (data: unknown) => string };
+  if (indexPattern.fieldFormatMap && indexPattern.fieldFormatMap[field.name]) {
+    const FormatType = fieldFormats.getType(indexPattern.fieldFormatMap[field.name].id);
+    formatter = new FormatType(
+      indexPattern.fieldFormatMap[field.name].params,
+      core.uiSettings.get.bind(core.uiSettings)
+    );
+  } else {
+    formatter = fieldFormats.getDefaultInstance(field.type, field.esTypes);
+  }
+
+  function fetchData() {
+    if (
+      state.isLoading ||
+      (field.type !== 'number' && field.type !== 'string' && field.type !== 'date')
+    ) {
+      return;
+    }
+
+    setState(s => ({ ...s, isLoading: true }));
+
+    core.http
+      .post(`/api/lens/index_stats/${indexPattern.title}/field`, {
+        body: JSON.stringify({
+          query: toElasticsearchQuery(query, indexPattern),
+          fromDate: dateRange.fromDate,
+          toDate: dateRange.toDate,
+          timeFieldName: indexPattern.timeFieldName,
+          field,
+        }),
+      })
+      .then((results: FieldStatsResponse<string | number>) => {
+        setState(s => ({
+          ...s,
+          isLoading: false,
+          totalDocuments: results.totalDocuments,
+          sampledDocuments: results.sampledDocuments,
+          sampledValues: results.sampledValues,
+          histogram: results.histogram,
+          topValues: results.topValues,
+        }));
+
+        if (
+          results.histogram &&
+          results.histogram.buckets.length &&
+          results.topValues &&
+          results.topValues.buckets.length
+        ) {
+          const count = results.sampledValues || results.sampledDocuments || results.totalDocuments;
+          const totalValuesCount =
+            results.topValues &&
+            results.topValues.buckets.reduce((prev, bucket) => bucket.count + prev, 0);
+          const otherCount = count && totalValuesCount ? count - totalValuesCount : 0;
+          // Default to histogram when top values are less than 10% of total
+          setShowingHistogram(otherCount / totalValuesCount > 0.9);
+        } else {
+          setShowingHistogram(!!results.histogram);
+        }
+      })
+      .catch(() => {
+        setState(s => ({ ...s, isLoading: false }));
+      });
+  }
+
+  const specId = getSpecId(
+    i18n.translate('xpack.lens.indexPattern.fieldStatsCountLabel', {
+      defaultMessage: 'Count',
+    })
+  );
+  const expectedColor = getColorForDataType(field.type);
+  const colors: DataSeriesColorsValues = {
+    colorValues: [],
+    specId,
+  };
+  const seriesColors = new Map([[colors, expectedColor]]);
+
+  const bucketedValuesCount =
+    state.topValues && state.topValues.buckets.reduce((prev, bucket) => bucket.count + prev, 0);
+  const otherCount =
+    state.sampledValues && bucketedValuesCount ? state.sampledValues - bucketedValuesCount : 0;
+
+  const euiButtonColor =
```

Comment:
These two values are identical. Is this incidental or inherent? If the latter, we should consolidate into one value.

Reply:
They're unfortunately not identical, buttons don't have a 'default' style and text doesn't have 'primary'